src/main/java/thewaypointers/trafficsimulator/TrafficSimulatorManager.java

```diff
@@ -25,17 +33,25 @@ public static void setMainFrame(MainFrame mainFrame) {
         TrafficSimulatorManager.mainFrame = mainFrame;
     }
 
+    public void setStart_pauseButton(JButton start_pauseButton) {
+        this.start_pauseButton = start_pauseButton;
+    }
     public TrafficSimulatorManager(JButton start_pause){
         start_pauseButton = start_pause;
     }
 
-    public void changeState() {
+
+
+    public synchronized void changeState() {
         isSleep = !isSleep;
         start_pauseButton.setText("Start");
     }
 
-    public void stopState() {
+    public synchronized void stopState() {
         stop = !stop;
+        simulation = new RoadNetworkProvider();
```

Comment:
Why does it create `RoadNetworkProvider`? We want to use it with other providers as well